src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -72,27 +72,31 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
     webapp_def = Site(location=location, site_config=site_config, server_farm_id=plan_info.id, tags=tags)
     helper = _StackRuntimeHelper(client, linux=is_linux)
 
-    if is_linux:
-        if not validate_linux_create_options(runtime, deployment_container_image_name,
-                                             multicontainer_config_type, multicontainer_config_file):
+    if is_linux or plan_info.is_xenon:
+        if not validate_container_app_create_options(runtime, deployment_container_image_name,
+                                                     multicontainer_config_type, multicontainer_config_file):
             raise CLIError("usage error: --runtime | --deployment-container-image-name |"
                            " --multicontainer-config-type TYPE --multicontainer-config-file FILE")
-        if startup_file:
+        if is_linux and startup_file:
             site_config.app_command_line = startup_file
 
-        if runtime:
+        if is_linux and runtime:
             site_config.linux_fx_version = runtime
             match = helper.resolve(runtime)
             if not match:
                 raise CLIError("Linux Runtime '{}' is not supported."
                                "Please invoke 'list-runtimes' to cross check".format(runtime))
         elif deployment_container_image_name:
-            site_config.linux_fx_version = _format_linux_fx_version(deployment_container_image_name)
+            fx_version = _format_fx_version(deployment_container_image_name)
+            if is_linux:
```

Comment:
remember, this condition will never be hit because the previous `if` has captured all linux case

Reply:
Hi @yugangw-msft - I think this condition can still be met if `runtime` is not present in the previous if (line 83)